### ISSUE TITLE
Difference in loading times between `lsdb.read_hipscat` and pyarrow.dataset.to_table 

### ISSUE THREAD

**Issue by nevencaplar** [astronomy-commons/lsdb#127]
@troyraen report that loading CosmoDC2 had large difference in loading times between `lsdb.read_hipscat` and pyarrow.dataset.to_table (see figure below). This is with ADASS tutorial branch.

The definitions are: `` defining "hot" as when the data was last accessed less than 10 minutes prior; "warm" 15min - 1hr prior; and "cold" about 24 hours prior''

![hipscat-pyarrow-load-times](https://github.com/astronomy-commons/lsdb/assets/12860669/98a93ba9-1bbe-46e6-80fe-de1ed70c6470)

We have been unable to recreate this issue. Potentially this is a feature of CosmoDC2 dataset which was not tested yet. Making this ticket to capture this investigation that is turning out somewhat bigger than a simple fix. Same notebook as specified in https://github.com/astronomy-commons/lsdb/issues/121

**Comment by troyraen** [comment]
Interesting that you were unable to reproduce. I suppose there could also be differences in our system setup that affect this. I also wouldn't rule out that it was just some combination of small-number stats and inconsistency across runs on my part. (I kept starting another run because the results kept being curious, but I didn't start out intending to do rigorous testing or record keeping.)

We're planning to regenerate the CosmoDC2 catalog using the new version of hipscat-import, and then we can also update the notebook this code came from to use the new lsdb. I can put the data in an S3 bucket where we can both test against it as soon as I have it, if that's helpful.

**Comment by hombit** [comment]
Hello @troyraen, thank you so much for the report!

I've tried to reproduce the issue, but with the different catalog and, therefore, different filters applied. It would be really helpful to have the same catalog you use. I can hipscat-import myself, if you point me to the original dataset you use.

**Comment by troyraen** [comment]
Hi @hombit, unfortunately the files used to make this CosmoDC2 hipscat catalog are not publicly available. I checked with some folks internally about releasing them, but ran into some issues. I'm fairly certain I can make them available to you privately for testing. If you're interested, email me (raen[at]ipac.caltech.edu) and I can give you more info about the files and figure out how to transfer.

**Comment by hombit** [comment]
Thank you @troyraen, I've just written you privately

**Comment by nevencaplar** [comment]
@hombit Can you shortly summarize the investigation here? I believe that we concluded to close this one without resolving at this stage, as it is due to different best defaults for lsdb and pyarrow?

**Comment by hombit** [comment]
@nevencaplar yes, I believe it is correct. I've found that in both cases of LSDB/Dask and pyarrow one could optimize number of parallel jobs to go really close to the performance limited by the read-out speed. It doesn't mean that we cannot do better : we can and hopefully will do after migration to `dask-expr`
